### PR TITLE
Add new tfx extension install command

### PR DESCRIPTION
**Pull request by bfcamara**
Add a new tfx extension command to install an extension in a list of accounts

### CODE REVIEW

File: app/exec/extension/install.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+
+    /**
+     * Determine the Accounts URL based on the service URL.
+     * TODO: Check if it is possible to do it using the REST API.
+     *
+     * @param  {string} serviceUrl
+     * @returns string
+     */
+    private _getAccountsServiceBaseUrl(serviceUrl: string): string {
```
Comment (bfcamara) [inline app/exec/extension/install.ts]: @trevorsg Probably we can do this in a better way.

File: app/exec/extension/install.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+/**
+ * TODO: Remove this as soon as it is included in the vso-node-api module
+ *
+ */
+
+import VsoBaseInterfaces = require("vso-node-api/interfaces/common/VsoBaseInterfaces");
+import { ClientApiBase } from  "vso-node-api/ClientApiBases";
+
+class AccountsApi extends ClientApiBase {
```
Comment (bfcamara) [inline app/exec/extension/install.ts]: @trevorsg While AccountsApi is not included vso-node-api this is the simplest way to get the list of accounts by member id.

**Comment by bfcamara** [comment]
Hi @trevorsg Could you review this PR please?

I left some notes with the changed files :

- we need to determine the Accounts (SPS) base URL in order to determine the accountId for a given accountName (The Acquisition API receives the accountId). I think the I am determining the SPS URL is not the best. Any idea is appreciated.

- the AccountsApi client is not included yet in the vso-node-api module. I'll work on a new PR to get the code generated for that API and make a new PR there. But for now, I would like to move forward with custom code to make the call on the Accounts API.

File: app/exec/extension/install.ts [hunk 3]
```diff
@@ -0,0 +1,489 @@
+import { TfCommand } from "../../lib/tfcommand";
+import args = require("../../lib/arguments");
+import colors = require("colors");
```
Comment (trevorsg) [inline app/exec/extension/install.ts]: Remove unused import

File: app/exec/extension/install.ts [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+export class AccountInstallReport {
+    constructor(
+        public itemId: string,
+        public accountName: string,
+        public accountId: string,
+        public installed: boolean = false,
+        public reason?: string) {
+    }
+
+    setError(reason: string) {
```
Comment (trevorsg) [inline app/exec/extension/install.ts]: public

File: app/exec/extension/install.ts [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+        this.installed = false;
+        this.reason = reason;
+    }
+
+    setInstalled(reason?: string) {
```
Comment (trevorsg) [inline app/exec/extension/install.ts]: public

File: app/exec/extension/install.ts [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+        return result;
+    }
+}
+
+export class ExtensionInstall extends extBase.ExtensionBase<ExtensionInstallResult> {
+    protected description = "Install a Visual Studio Services Extension in a list of VSTS Accounts.";
+
+    constructor(passedArgs: string[]) {
+        super(passedArgs);
+        this.registerCommandArgument("accounts", "Installation target accounts", "List of accounts where to install the extension.", args.ArrayArgument);
```
Comment (trevorsg) [inline app/exec/extension/install.ts]: Override setCommandArgs(), call super.setCommandArgs(), then register this new one.

File: app/exec/extension/install.ts [hunk 7]
```diff
@@ -0,0 +1,489 @@
...
+        this.accountReports.forEach(t => {
+            result += `\n    - ${t.accountName}: [${t.installed ? "OK" : `Failed: ${t.reason}`}]`;
+        });
+
+        return result;
+    }
+}
+
+export class ExtensionInstall extends extBase.ExtensionBase<ExtensionInstallResult> {
+    protected description = "Install a Visual Studio Services Extension in a list of VSTS Accounts.";
```
Comment (trevorsg) [inline app/exec/extension/install.ts]: Install a Visual Studio Services Extension TO a list of VSTS Accounts.

File: app/exec/extension/install.ts [hunk 8]
```diff
@@ -0,0 +1,489 @@
+import { TfCommand } from "../../lib/tfcommand";
```
Comment (trevorsg) [inline app/exec/extension/install.ts]: Use tabs to match the rest of the files' indentation method. If you are passionate about using spaces, let's fight about it!

**Pull request merged**